api/pkg/s3/api-response.go

```diff
@@ -194,4 +201,38 @@ func parseListUploadsQuery(query url.Values) (request ListUploadsRequest, err er
 	request.Prefix = query.Get("prefix")
 	request.UploadIdMarker = query.Get("upload-id-marker")
 	return
-}
\ No newline at end of file
+}
+
+// Parse object url queries
+func parseListObjectPartsQuery(query url.Values) (request ListPartsRequest, err error) {
+	request.EncodingType = query.Get("encoding-type")
+	request.UploadId = query.Get("uploadId")
+	if request.UploadId == "" {
+		err = ErrNoSuchUpload
+		return
+	}
+	if query.Get("max-parts") == "" {
```

Comment:
if its null, we default to MaxPartsList ? -- is this correct? Because other invalid checks are handled below.

Reply:
I thinks this max-parts is used to sets the maximum number of parts to return, if the user did not set it, then we use a default value, for example, we return 1000 parts as default.

Reply:
@skdwriting  thanks for your reviewing,  if max-parts is not specified by user, we will use MaxPartsList as default.